application/src/main/java/org/togetherjava/tjbot/commands/moderation/report/ReportCommand.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        long userID = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userID, event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+        event.deferReply().setEphemeral(true).queue();
+
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), guildID);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+            return;
+        }
+        String modalMessage = event.getValue(MESSAGE_INPUT).getAsString();
+        MessageCreateAction message =
+                createModMessage(modalMessage, userID, modMailAuditLog.orElseThrow());
+
+        sendMessage(event, message);
+
+        event.getHook().editOriginal("Thank you for your feedback!").queue();
+
+    }
+
+    private boolean handleIsOnCooldown(long userID, ModalInteractionEvent event) {
+        if (!isChannelOnCooldown(userID)) {
+            return false;
+        }
+        event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isChannelOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private Optional<TextChannel> getModMailChannel(JDA jda, long guildID) {
+        return jda.getGuildById(guildID)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private MessageCreateAction createModMessage(String modalMessage, long userID,
+            TextChannel modMailAuditLog) {
+        return modMailAuditLog.sendMessageEmbeds(createReportMessage(modalMessage, userID));
+    }
+
+    private void sendMessage(ModalInteractionEvent event, MessageCreateAction message) {
+        InteractionHook hook = event.getHook();
+        message.mapToResult().map(result -> {
+            if (result.isSuccess()) {
+                return "Your message has been forwarded, thanks.";
+            }
+            logger.warn("There was an issue with forwarding users message.");
+            return "There was an issue forwarding your message, sorry. We are investigating.";
+        }).flatMap(hook::editOriginal).queue();
+    }
```

Comment:
okay, now i am confused. you already edited the original before, saying "thank you for your feedback". one of them is definitely a mistake and should be removed.

in either case, everything i said at the other, also applies here. the message should be improved

Reply:
Done. I removed the eventHook that comes after it